File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/attivita/centri_di_costo/CapoAnimaleCDC.cs
using System;
using System.Collections.Generic;
using System.Text;
using AgronicaCoreModelsSTD.anagrafiche;
using AgronicaCoreModelsSTD.documenti;

namespace AgronicaCoreModelsSTD.attivita.centri_di_costo
{
    /// <summary>
    /// 
    /// </summary>
    public class CapoAnimaleCDC : CentroDiCosto
    {
        #region Properties

        public SottogruppoStalla sottogruppoStalla_ingresso { get; set; }

        public SottogruppoStalla sottogruppoStalla_uscita { get; set; }

        public CapoAnimale capoAnimale { get; set; }

        public baseClass.BaseCodeDescr capoAnimaleNonPresente { get; set; }

        public int id_movimentazione_BDN { get; set; }

        public double qtaSomministrata { get; set; }

        /// <summary>
        /// Indica se il Capo sta venendo trattato con antibiotici.
        /// </summary>
        public bool onAntibiotico { get; set; }
        /// <summary>
        /// Indica se il Capo sta venendo trattato con antinfiammatori.
        /// </summary>
        public bool onAntinfiammatorio { get; set; }

        #endregion

        #region Constructors

        public CapoAnimaleCDC()
        {
            classType = costanti.ClassType.CapoAnimaleCDC;
            tipo = Tipo.CapoAnimale;
        }

        #endregion

    }
}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/attivita/centri_di_costo/CapoAnimaleCDC.ts
import { BaseCodeDescr, CapoAnimale, SottogruppoStalla } from "app/Service/api.service";
import { CentroDiCosto, Tipo } from "./CentroDiCosto";

export class CapoAnimaleCDC extends CentroDiCosto {

    sottogruppoStalla_ingresso: SottogruppoStalla;

    sottogruppoStalla_uscita: SottogruppoStalla;

    capoAnimale: CapoAnimale;

    capoAnimaleNonPresente: BaseCodeDescr;

    id_movimentazione_BDN: number;

    qtaSomministrata: number;

    constructor() {
        super();
        this.classType = "CapoAnimaleCDC";
        this.tipo = Tipo.CapoAnimale;
    }

}